libraries/provider_service.rb

```diff
@@ -0,0 +1,67 @@
+class Chef
+  # Chef Provider for configuring an elasticsearch service in the init system
+  class Provider::ElasticsearchService < Provider
+    include Poise
+
+    def action_remove
+    end
+
+    def action_configure
+      directory new_resource.pid_path do
+        mode '0755'
+        recursive true
+      end
+
+      # Create service
+      #
+      template "/etc/init.d/#{new_resource.service_name}" do
+        source "elasticsearch.init.erb"
+        cookbook 'elasticsearch'
+        owner 'root'
+        mode 0755
+        variables({
+          nofile_limit: new_resource.nofile_limit,
+          memlock_limit: new_resource.memlock_limit,
+          pid_file: new_resource.pid_file,
+          path_conf: new_resource.path_conf,
+          user: new_resource.user,
+          platform_family: node.platform_family,
+          bindir: new_resource.bindir,
+          http_port: 9200, # TODO: does the init script really need this?
+          node_name: new_resource.node_name,
+          service_name: new_resource.service_name
+          })
+      end
+
+      # Increase open file and memory limits
+      #
+      bash "enable user limits" do
```

Comment:
We might need not to fiddle with system limits, if we do what the init scripts for deb/rpm packages do, and that is [call `ulimit`](https://github.com/elastic/elasticsearch/blob/master/src/packaging/deb/init.d/elasticsearch#L154) during the startup process?